### PR TITLE
add providers: zookeeper、kafka、elasticsearch、etcd、etcd-mutex …

### PR DESCRIPTION
**Pull request by recallsong**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

add providers: zookeeper、kafka、elasticsearch、etcd、etcd-mutex、zk-master-election、kubernetes.

#### Special notes for your reviewer:
```
```

#### Reivewers List:
@liuhaoyang

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/parallel-writer/buffer.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+package writer
+
+import (
+	"github.com/recallsong/go-utils/errorx"
+)
+
+// Buffer .
+type Buffer struct {
+	w       Writer
+	buf     []interface{}
```
Comment (liuhaoyang) [inline pkg/parallel-writer/buffer.go:13]: single-letter field names are not allowed

File: providers/etcd-mutex/mutex.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+}
+
+type config struct {
+	RootPath   string `file:"root_path"`
+	DefaultKey string `file:"default_key"`
+}
+
+type provider struct {
+	C           *config
+	L           logs.Logger
```
Comment (liuhaoyang) [inline providers/etcd-mutex/mutex.go:62]: bad naming

**Review by liuhaoyang — CHANGES_REQUESTED** [review]

File: pkg/parallel-writer/buffer.go (hunk 1 above)
Comment (sfwn) [reply pkg/parallel-writer/buffer.go:13]: `w` I think is good.

File: providers/elasticsearch/elasticsearch.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+var clientType = reflect.TypeOf((*elastic.Client)(nil))
+
+type config struct {
+	URLs     string `file:"urls" default:"http://localhost:9200" desc:"servers urls"`
+	Security bool   `file:"security" default:"false" desc:"enable http basic auth"`
+	Username string `file:"username" default:"" desc:"username"`
+	Password string `file:"password" default:"" desc:"password"`
+}
```
Comment (liuhaoyang) [inline providers/elasticsearch/elasticsearch.go:43]: Add reading configuration from envs

**Review by liuhaoyang — CHANGES_REQUESTED** [review]

**Review by liuhaoyang — APPROVED** [review]

**Pull request merged**